include/ozo/impl/write.h

```diff
@@ -0,0 +1,72 @@
+#pragma once
+
+#include <ozo/concept.h>
+#include <ozo/error.h>
+#include <ozo/detail/ostream.h>
+#include <ozo/detail/endian.h>
+#include <ozo/detail/float.h>
+#include <ozo/detail/typed_buffer.h>
+#include <boost/hana/for_each.hpp>
+#include <boost/hana/members.hpp>
+#include <boost/hana/tuple.hpp>
+
+namespace ozo::impl {
+
+using detail::ostream;
+
+template <typename T>
+inline Require<Integral<T> && sizeof(T) == 1, ostream&>
+write(ostream& out, T in) {
```

Comment:
Well, to be honest, I don't remember any conventions about this. At least we have same style in "connection.h", "binary_deserealization.h" and so on. So if you have a suggestion to use this style of return type with Require statement emulation the best choice is to tell us why it is better and get a convention between all of us.

Reply:
I don't have any good reason for my suggestion now.

Reply:
For now I discovered the only one good reason - in case of something like this:

```c++
template <typename T1, typename T2>
inline auto foo(very_long<T1, and<sopthisticated_template_exprassion<T>>>& v) -> decltype(bar(v)) {
    return bar(v);
}
```